SPDX-FileCopyrightText: 2024 Roxy Light <roxy@zombiezen.com>

SPDX-License-Identifier: MIT

Ignore sticky/setuid/setgid bits during extraction
so such archives can be used in sandboxed builds.

--- src/extract.c
+++ src/extract.c
@@ -103,7 +103,7 @@
 
   if (!keep_old_files_option
       || (stat_info->st_mode & (S_ISUID | S_ISGID | S_ISVTX)))
-    if (chmod (file_name, ~current_umask & (int) stat_info->st_mode) < 0)
+    if (chmod (file_name, ~(current_umask | 07000) & (int) stat_info->st_mode) < 0)
       ERROR ((0, errno, _("%s: Cannot change mode to %0.4o"),
 	      file_name, ~current_umask & (int) stat_info->st_mode));
 }
